community/bolt/src/main/java/org/neo4j/bolt/transport/DefaultThrottleLock.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.bolt.transport;
+
+import io.netty.channel.Channel;
+
+public class DefaultThrottleLock implements ThrottleLock
+{
+    private final Object syncObject = new Object();
+
+    @Override
+    public void lock( Channel channel, long timeout )
+    {
+        synchronized ( syncObject )
+        {
+            try
+            {
+                syncObject.wait( timeout );
+            }
+            catch ( InterruptedException e )
+            {
+                // do nothing
```

Comment:
Interruption status should probably be restored with `Thread.currentThread().interrupt()`. Maybe we also want to re-throw some `RuntimeException` from here? Otherwise caller might proceed thinking it successfully locked.

Reply:
Thansk for pointing this out! I think it would be better to propagate `InterruptedException` to the caller and make it `clear` the interrupt flag and continue its loop.